apps/ejabberd/src/ejabberd_binary.erl

```diff
@@ -28,3 +29,16 @@ string_to_binary(S) when is_list(S) ->
     end;
 string_to_binary(B) when is_binary(B) ->
     B.
+
+
+join([], _) ->
```

Comment:
Maybe this could be more optimal:
```erlang
join(Parts, Sep) when is_integer(Sep) -> join(Parts, <<Sep>>);
join(Parts, Sep) -> join(Parts, Sep, <<>>).

join([], _Sep, Acc) -> Acc;
join([Part], _Sep, Acc) -> <<Acc/binary, Part/binary>>;
join([Part | RParts], Sep, Acc) -> join2(RParts, Sep, <<Acc/binary, Part/binary, Sep/binary>>).
```